app/controllers/course/survey/surveys_controller.rb

```diff
@@ -67,6 +74,10 @@ def download
 
   private
 
+  def number_of_students_in_course
+    @course_students = current_course.course_users.students
```

Comment:
I suppose this is needed only because the `_survey` partial needs it for `json.numberOfStudents`?

Anyway, this "pseudo-memoisation" will not help prevent repeated queries because `course_users.students` is still just an Active Record Relation. Every time one does `@course_students.count` (not to mention using `count` instead of `size`), one SQL query will be made. This means, for example in `#index`, you will do N exact same SQL queries to count the number of students in the course, where N = `@surveys.size`.

What you want in this case is to do cache the actual integer, not the relation.